In [1]:
import pandas as pd

In [2]:
vine_df = pd.read_csv('vine_table.csv')
vine_df

review_id  star_rating  helpful_votes  total_votes vine  \
0        R1WBPB8MDCCN8F          5.0            0.0          0.0    N   
1        R32M0YEWV77XG8          5.0            1.0          1.0    N   
2         RR8V7WR27NXJ5          1.0            0.0          0.0    N   
3        R1MHO5V9Z932AY          5.0            0.0          0.0    N   
4        R16PD71086BD2V          5.0            0.0          1.0    N   
...                 ...          ...            ...          ...  ...   
4850355  R1VERHFY08SLZS          1.0            6.0          8.0    N   
4850356  R34TPFL5ZYSV46          3.0           21.0         29.0    N   
4850357   R2YECHGBV291H          5.0            1.0          1.0    N   
4850358  R2Z784CWN7JZ4X          2.0            3.0          4.0    N   
4850359  R2J13W5P3N2KV4          4.0           18.0         19.0    N   

        verified_purchase  
0                       N  
1                       Y  
2                       Y  
3                       Y  
4                       N  
...                   ...  
4850355                 N  
4850356                 N  
4850357                 N  
4850358                 N  
4850359                 N  

[4850360 rows x 6 columns]

#### 1. Filter for rows where Total Votes are greater than or equal to 20

In [4]:
total_votes_20_or_more = (vine_df['total_votes'] >= 20)
vine_df_total_votes_filtered = vine_df.loc[total_votes_20_or_more]
vine_df_total_votes_filtered

review_id  star_rating  helpful_votes  total_votes vine  \
326      R1B4753QMTD8ON          5.0           32.0         32.0    N   
662      R28BVHOEL6GCHZ          5.0           21.0         21.0    N   
1055     R27IB4BRKO71I0          2.0           16.0         20.0    N   
1225      RJ86OW12Y4NRL          5.0          124.0        128.0    N   
1448     R3KRABOWCH6ZE4          1.0           48.0         65.0    N   
...                 ...          ...            ...          ...  ...   
4850347  R1KGFSPKL3YCGK          4.0           25.0         26.0    N   
4850349  R15AZWCW5ZSFSN          5.0          157.0        160.0    N   
4850352  R1ZWVHA89AE49B          5.0           32.0         33.0    N   
4850353  R115HJRM6FUVIC          5.0           27.0         27.0    N   
4850356  R34TPFL5ZYSV46          3.0           21.0         29.0    N   

        verified_purchase  
326                     N  
662                     Y  
1055                    Y  
1225                    Y  
1448                    Y  
...                   ...  
4850347                 N  
4850349                 Y  
4850352                 N  
4850353                 N  
4850356                 N  

[67855 rows x 6 columns]

#### 2. Filter the new DataFrame created in Step 1 and create a new DataFrame to retrieve all the rows where the number of helpful_votes divided by total_votes is equal to or greater than 50%.

In [5]:
helpful_votes_filter = (vine_df_total_votes_filtered['helpful_votes']/vine_df_total_votes_filtered['total_votes'] >= 0.5)
vine_df_helpful = vine_df_total_votes_filtered.loc[helpful_votes_filter]
vine_df_helpful

review_id  star_rating  helpful_votes  total_votes vine  \
326      R1B4753QMTD8ON          5.0           32.0         32.0    N   
662      R28BVHOEL6GCHZ          5.0           21.0         21.0    N   
1055     R27IB4BRKO71I0          2.0           16.0         20.0    N   
1225      RJ86OW12Y4NRL          5.0          124.0        128.0    N   
1448     R3KRABOWCH6ZE4          1.0           48.0         65.0    N   
...                 ...          ...            ...          ...  ...   
4850347  R1KGFSPKL3YCGK          4.0           25.0         26.0    N   
4850349  R15AZWCW5ZSFSN          5.0          157.0        160.0    N   
4850352  R1ZWVHA89AE49B          5.0           32.0         33.0    N   
4850353  R115HJRM6FUVIC          5.0           27.0         27.0    N   
4850356  R34TPFL5ZYSV46          3.0           21.0         29.0    N   

        verified_purchase  
326                     N  
662                     Y  
1055                    Y  
1225                    Y  
1448                    Y  
...                   ...  
4850347                 N  
4850349                 Y  
4850352                 N  
4850353                 N  
4850356                 N  

[61948 rows x 6 columns]

#### 3. Filter the DataFrame or table created in Step 2, and create a new DataFrame or table that retrieves all the rows where a review was written as part of the Vine program (paid), vine == 'Y'.

In [6]:
vine_program_filter = (vine_df_helpful['vine'] == 'Y')
vine_df_paid_review = vine_df_helpful.loc[vine_program_filter]
vine_df_paid_review

review_id  star_rating  helpful_votes  total_votes vine  \
5506     R32OMS1LSDRCOB          4.0           27.0         31.0    Y   
24578    R2IPKK4I0XRCFR          4.0           16.0         21.0    Y   
58364     RXYYJ8SGI9IX6          5.0           36.0         41.0    Y   
63194    R1SAUBFC7T6JHC          5.0           73.0         75.0    Y   
177724   R329HVJKKHWSG3          5.0           59.0         59.0    Y   
...                 ...          ...            ...          ...  ...   
4717565  R1WG2F3DZFE301          5.0           51.0         81.0    Y   
4719313  R3VB53V1VDWPQ2          4.0          104.0        132.0    Y   
4754150   RCKH2IYVH9PCG          4.0           21.0         21.0    Y   
4770385  R1OR7JY4MQP6I7          5.0          181.0        188.0    Y   
4770596  R34IO2BR3JWIIZ          4.0           17.0         20.0    Y   

        verified_purchase  
5506                    N  
24578                   N  
58364                   N  
63194                   N  
177724                  N  
...                   ...  
4717565                 N  
4719313                 N  
4754150                 N  
4770385                 N  
4770596                 N  

[334 rows x 6 columns]

#### 4. Repeat Step 3, but this time retrieve all the rows where the review was not part of the Vine program (unpaid), vine == 'N'.

In [7]:
vine_non_program_filter = (vine_df_helpful['vine'] == 'N')
vine_df_unpaid_review = vine_df_helpful.loc[vine_non_program_filter]
vine_df_unpaid_review

review_id  star_rating  helpful_votes  total_votes vine  \
326      R1B4753QMTD8ON          5.0           32.0         32.0    N   
662      R28BVHOEL6GCHZ          5.0           21.0         21.0    N   
1055     R27IB4BRKO71I0          2.0           16.0         20.0    N   
1225      RJ86OW12Y4NRL          5.0          124.0        128.0    N   
1448     R3KRABOWCH6ZE4          1.0           48.0         65.0    N   
...                 ...          ...            ...          ...  ...   
4850347  R1KGFSPKL3YCGK          4.0           25.0         26.0    N   
4850349  R15AZWCW5ZSFSN          5.0          157.0        160.0    N   
4850352  R1ZWVHA89AE49B          5.0           32.0         33.0    N   
4850353  R115HJRM6FUVIC          5.0           27.0         27.0    N   
4850356  R34TPFL5ZYSV46          3.0           21.0         29.0    N   

        verified_purchase  
326                     N  
662                     Y  
1055                    Y  
1225                    Y  
1448                    Y  
...                   ...  
4850347                 N  
4850349                 Y  
4850352                 N  
4850353                 N  
4850356                 N  

[61614 rows x 6 columns]

Check:


Number of rows from vine_df_paid_review  + Number of rows from vine_df_unpaid_review = Number of rows from vine_df_helpful


334 + 66,614 = 66,948

#### 5. Determine the total number of reviews, the number of 5-star reviews, and the percentage of 5-star reviews for the two types of review (paid vs unpaid).

##### Paid Reviews

In [14]:
vine_df_paid_review['star_rating'].count()

334

In [12]:
vine_df_paid_review['star_rating'].value_counts()

5.0    139
4.0    118
3.0     56
2.0     13
1.0      8
Name: star_rating, dtype: int64

In [13]:
vine_df_paid_review['star_rating'].value_counts(normalize=True)

5.0    0.416168
4.0    0.353293
3.0    0.167665
2.0    0.038922
1.0    0.023952
Name: star_rating, dtype: float64

##### Unpaid Reviews

In [15]:
vine_df_unpaid_review['star_rating'].count()

61614

In [16]:
vine_df_unpaid_review['star_rating'].value_counts()

5.0    32665
4.0    10688
1.0     9192
3.0     5477
2.0     3592
Name: star_rating, dtype: int64

In [17]:
vine_df_unpaid_review['star_rating'].value_counts(normalize=True)

5.0    0.530155
4.0    0.173467
1.0    0.149187
3.0    0.088892
2.0    0.058298
Name: star_rating, dtype: float64

#### Results

Percentage of 5-star rating for paid reviews: 42%

Percentage of 5-star rating for unpaid reviews: 53%

Having a paid Vine review does not appear to make a difference in the percentage of 5-star reviews. In fact, the percentage of 5-star ratings from paid reviews is 11% lower than unpaid reviews.